### PR TITLE
Step 3 & 4

### PR DESCRIPTION
**Pull request by Kwameoab**
- Get Json data from server
- Run prettier
- Add data to server through Post

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -19,11 +19,15 @@
...
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+
```
Comment (rayliu0317) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Remove the empty lines.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    response.getWriter().println(json);
+  }
+
+  private String convertToJson(ArrayList<String> comments){
+      String json = "{";
+      for(int i=0; i < comments.size(); ++i){
+          if (i != 0){
+              json += ", ";
+          }
+          json += "\"Comment " + (i+1) + "\": ";
```
Comment (rayliu0317) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java]: space around +

(i + 1)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    String json = convertToJson(comments);
+
+    response.setContentType("application/json;");
+    response.getWriter().println(json);
+  }
+
+  private String convertToJson(ArrayList<String> comments){
+      String json = "{";
+      for(int i=0; i < comments.size(); ++i){
```
Comment (rayliu0317) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java]: space around =

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import com.google.gson.Gson;
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/jData")
```
Comment (rayliu0317) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java]: this name can be confusing: what does jData stands for exactly? jsonData? javaData?
Try to make it more explicit and descriptive.

File: portfolio/src/main/webapp/index.html [hunk 5]
```diff
@@ -60,8 +60,25 @@ <h1>Kwame Ampadu-Boateng's Portfolio</h1>
       </p>
       <button onclick="addRandomGreeting()">Explore</button>
       <div id="greeting-container"></div>
-      <button onclick="getData()">Fetch a quote!</button>
+
+      <br />
```
Comment (rayliu0317) [inline portfolio/src/main/webapp/index.html:64]: I think <br> is sufficient if you only want a newline (line-break).

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/jsonData")
+public class DataServlet2 extends HttpServlet {
+    private ArrayList<String> comments = new ArrayList();
+
+
```
Comment (rayliu0317) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java]: Is these empty lines intentional? one should be enough most of the times.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/jsonData")
+public class DataServlet2 extends HttpServlet {
+    private ArrayList<String> comments = new ArrayList();
+
+  
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
```
Comment (rayliu0317) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java]: Remove empty line here.

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -34,23 +34,39 @@ function addRandomGreeting() {
...
+  console.log("We are Here;");
+  const responsePromise = fetch("/data");
 
-    responsePromise.then(handleResponse);
+  responsePromise.then(handleResponse);
 }
 
 function handleResponse(response) {
-  console.log('Handling the response.');
+  console.log("Handling the response.");
```
Comment (rayliu0317) [inline portfolio/src/main/webapp/script.js:44]: I noticed that you are constantly changing the use of single quote to double quote. FYI, they are equivalent in terms of representing a string in MOST cases: https://staxmanade.com/2018/03/should-i-use-javascript-single-or-double-quotes/

Which one to choose? Well doesn't really matter, just keep consistent within the file/library.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+          if (i != 0){
+              json += ", ";
+          }
+          json += "\"Comment " + (i + 1) + "\": ";
+          json += "\"" + comments.get(i) + "\"";
+      }
+      json += "}";
+      return json;
+  }
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```
Comment (rayliu0317) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet2.java:54]: newline between functions.

File: portfolio/src/main/webapp/script.js (hunk 8 above)
Comment (Kwameoab) [reply portfolio/src/main/webapp/script.js:44]: It is not me who changes the quotes, I am using the program Lyle told us about to format our code and for some reason it likes to change the quotation marks.

File: portfolio/src/main/webapp/index.html (hunk 5 above)
Comment (Kwameoab) [reply portfolio/src/main/webapp/index.html:64]: Sorry, I am confused by this comment.

Comment (rayliu0317) [reply portfolio/src/main/webapp/index.html:64]: Ooops I typed ```<br>``` and obviously it was automatically interpreted as a line-break.

Comment (Kwameoab) [reply portfolio/src/main/webapp/index.html:64]: I just checked this out. This is also a strange thing prettier does, it changes all of my `<br>` to `<br />`

**Review by rayliu0317 — APPROVED** [review]

**Pull request merged**